Copied from
https://github.com/gnif/LookingGlass/issues/1075#issuecomment-1546422678 and
https://github.com/gnif/LookingGlass/issues/1134 with adjustments for current
kvmfr source version.

From c4950a830fbe2ca27337793aa227c86f5c044f46 Mon Sep 17 00:00:00 2001
From: Oleg Pykhalov <go.wigust@gmail.com>
Date: Sat, 5 Oct 2024 16:11:45 +0300
Subject: [PATCH] Fix build

---
 module/kvmfr.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/module/kvmfr.c b/module/kvmfr.c
index 121aae5..4c386f9 100644
--- a/module/kvmfr.c
+++ b/module/kvmfr.c
@@ -30,6 +30,7 @@
 #include <linux/highmem.h>
 #include <linux/memremap.h>
 #include <linux/version.h>
+#include <linux/vmalloc.h>
 
 #include <asm/io.h>
 
@@ -539,7 +540,11 @@ static int __init kvmfr_module_init(void)
   if (kvmfr->major < 0)
     goto out_free;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0)
   kvmfr->pClass = class_create(THIS_MODULE, KVMFR_DEV_NAME);
+#else
+  kvmfr->pClass = class_create(KVMFR_DEV_NAME);
+#endif
   if (IS_ERR(kvmfr->pClass))
     goto out_unreg;
 
-- 
2.41.0

